client/state/data-layer/wpcom/sites/automated-transfer/initiate/index.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import { initiateTransfer, requestEligibility } from 'state/automated-transfer/actions';
+import { transferStates } from 'state/automated-transfer/constants';
+import {
+	AUTOMATED_TRANSFER_INITIATE_REQUEST,
+} from 'state/action-types';
+
+/**
+ * Maps from API response to internal representation of automated transfer initialize request
+ *
+ * @param {Object} data API response data
+ * @returns {Object} Calypso eligibility information
+ */
+const fromApi = data => Object.assign(
+	{
+		id: data.transfer_id,
+	},
+	data.status === 'active' && { status: transferStates.START },
+	data.status === 'uploading' && { status: transferStates.START },
+	data.active_theme_slug && data.active_theme_slug.length && { theme: { id: data.active_theme_slug } },
+);
+
+/**
+ * Dispatches to initiate an automated transfer for a site
+ *
+ * @param {Function} dispatch action dispatcher
+ * @param {number} siteId site for which the update belongs
+ * @returns {Function} the handler function with site and dispatch partially applied
+ */
+const apiResponse = ( dispatch, siteId ) => data => dispatch( initiateTransfer( siteId, fromApi( data ) ) );
+
+/**
+ * Respond to API fetch failure
+ *
+ * @param {Object} error error from API fetch
+ */
+const apiFailure = ( dispatch, siteId ) => () => dispatch( requestEligibility( siteId ) );
+
+/**
+ * Issues an API request to initiate an automatic transfer
+ *
+ * @param {Function} dispatch action dispatcher
+ * @param {number} siteId which site to transfer
+ * @param {string} pluginSlug name of community plugin initiating transfer
+ * @param {File} themeFile zip of theme initiating transfer
+ * @returns {Promise} response promise from API fetch
+ */
+export const requestTransfer = ( { dispatch }, { siteId, pluginSlug, themeFile } ) =>
```

Comment:
@seear this hasn't hit all of Calypso yet as it's very new and a few of us have been using it to work out the bugs. are you planning on doing any more work with that code in `undocumented`? if not, I would be willing to move in after this PR merges to move your code and tests over to this new system. the good side is that units tests become significantly easier.

if not, I'd probably prefer to keep them duplicated in there for now but am willing to give this up if we have to.